riklet/src/cli/function_config.rs

```diff
@@ -7,24 +9,34 @@ use clap::Parser;
 pub struct FnConfiguration {
     pub firecracker_location: PathBuf,
     pub kernel_location: PathBuf,
-}
 
-impl From<CliConfiguration> for FnConfiguration {
-    fn from(cli: CliConfiguration) -> Self {
-        FnConfiguration {
-            firecracker_location: cli.firecracker_path,
-            kernel_location: cli.kernel_path,
-        }
-    }
+    /// IP used to access the external network
+    pub gateway_ip: Ipv4Addr,
+    /// Network interface used to access the external network
+    pub iface: String,
 }
 
 impl FnConfiguration {
     fn get_cli_args() -> CliConfiguration {
         CliConfiguration::parse()
     }
 
-    pub fn load() -> Self {
+    pub fn load() -> Result<FnConfiguration, anyhow::Error> {
         let opts = FnConfiguration::get_cli_args();
-        FnConfiguration::from(opts)
+
+        let gateway_ip = match opts.iface_ip {
+            Some(ip) => ip,
+            None => get_default_gateway()?,
+        };
+        let iface = match opts.iface {
+            Some(iface) => iface,
+            None => get_default_iface()?,
```

Comment:
🟢 maybe anyhow to NetworkError ? 
hummm i don't know

Reply:
NetworkError is not related to an error given by a third party library, and as stated before, load of configuration (and of iface / gateway) is done so early that we can't handle errors and have a different behavior, that's why anyhow is used.

In general, I don't think we should be using proper Error type when we can't handle them properly with various cases.